pkg/providers/v1/delete_tags_batch.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"github.com/mitchellh/hashstructure/v2"
+	batcher2 "k8s.io/cloud-provider-aws/pkg/providers/v1/batcher"
+	"k8s.io/cloud-provider-aws/pkg/providers/v1/iface"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+// DeleteTagsBatcher contains the batcher details
+type DeleteTagsBatcher struct {
+	batcher *batcher2.Batcher[ec2.DeleteTagsInput, ec2.DeleteTagsOutput]
+}
+
+// NewDeleteTagsBatcher creates a NewDeleteTagsBatcher object
+func NewDeleteTagsBatcher(ctx context.Context, ec2api iface.EC2) *DeleteTagsBatcher {
```

Comment:
These should be package-private
```suggestion
type deleteTagsBatcher struct {
	batcher *batcher2.Batcher[ec2.DeleteTagsInput, ec2.DeleteTagsOutput]
}

func newDeleteTagsBatcher(ctx context.Context, ec2api iface.EC2) *DeleteTagsBatcher {
```